### PR TITLE
feature(lxy): add popart demo

### PR DESCRIPTION
**Pull request by karroyan**

### CODE REVIEW

File: chapter4_reward/popart_demo.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def train(obs_shape: int, action_shape: int, NUM_EPOCH: int, train_data):
+    model = MLP(obs_shape, action_shape)
+    optimizer = Adam(model.parameters(), lr=0.0001, weight_decay=0.0001)
+    MSEloss = nn.MSELoss()
+    # Read the preprocessed data of trained agent on lunarlander.
+    # Data format: list of dict with keys: ``{'observations', 'actions', 'rewards'}`` , 
+    # where the rewards is the discounted return from the current state. 
+    train_data = DataLoader(train_data, batch_size = 64, shuffle = True)
```
Comment (PaParaZz1) [inline chapter4_reward/popart_demo.py:52]: correct format

File: chapter4_reward/popart_demo.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        x = torch.cat((obs, actions),1)
+        x = self.encoder(x)
+        # The popart layer maps the embedding vector to a normalized value.
+        normalized_output  = self.popart(x)
+        return normalized_output 
+
+
+def train(obs_shape: int, action_shape: int, NUM_EPOCH: int, train_data):
+    model = MLP(obs_shape, action_shape)
+    optimizer = Adam(model.parameters(), lr=0.0001, weight_decay=0.0001)
```
Comment (PaParaZz1) [inline chapter4_reward/popart_demo.py:47]: AdamW

File: chapter4_reward/popart_demo.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                normalized_reward = (data['rewards'] - mu)/sigma
+            # The loss is calculated as the MSE loss between normalized Q value and normalized target return.
+            loss = MSEloss(normalized_output, normalized_reward)
+            loss.backward()
+            optimizer.step()
+            # After the model parameters are updated with the gradient, the weights and bias should be updated to preserve unnormalised outputs.
+            model.popart.update_parameters(data['rewards'])
+
+            running_loss = 0.0
+            running_loss += loss.item()
```
Comment (PaParaZz1) [inline chapter4_reward/popart_demo.py:72]: why not just use `loss.item()`

File: chapter4_reward/popart_demo.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        if epoch % 100 == 99:    
+            print('Epoch [%d] loss: %.6f' % (epoch + 1, running_loss / 100))
+            running_loss = 0.0
+            
+if __name__ == '__main__':
+    f = open('processed_data.pkl', 'rb')
```
Comment (PaParaZz1) [inline chapter4_reward/popart_demo.py:79]: use `with context`

Comment (PaParaZz1) [inline chapter4_reward/popart_demo.py:79]: move data to our official download website, i.e., https://opendilab.net/download/PPOxFamily/

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]
